.github/workflows/helm-chart-release.yml

```diff
@@ -82,3 +82,55 @@ jobs:
           publish_dir: ./to-gh-pages
           keep_files: true
           enable_jekyll: true
+  release-oci:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      packages: write
+    steps:
+      - name: Checkout source code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Install cosign
+        uses: sigstore/cosign-installer@main
+
+      - name: Install Helm
+        uses: azure/setup-helm@v1
+        with:
+          version: v3.8.0
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Login Helm to the registry
+        run: |
+          echo "${{ secrets.GITHUB_TOKEN }}" | helm registry login ghcr.io --username "${{ github.repository_owner }}" --password-stdin
```

Comment:
Isn't there the risk of leaking the contents of the token?

According to the [official docs](https://docs.github.com/en/actions/security-guides/automatic-token-authentication):

> Before each job begins, GitHub fetches an installation access token for the job. The GITHUB_TOKEN expires when a job finishes or after a maximum of 24 hours.

Leaking the token should not be a big deal, in theory...

Can't you authenticate using the docker login action? Is helm going to read the `~/.docker/config.json` file?

Reply:
I don't think this will print the token, and if you try to print the token github will replace it with ***. 
Having said that I also think it would be better to use the docker login action if possible